internal/endian_le.go

```diff
@@ -0,0 +1,10 @@
+//go:build amd64 || arm64
+// +build amd64 arm64
+
+package internal
+
+import "encoding/binary"
+
+var NativeEndian binary.ByteOrder = binary.LittleEndian
```

Comment:
Can you explain the purpose of hard coding the byte order for amd64?

Reply:
My thought was that it might help the compiler do more optimizations since it changes the determination from runtime to build time.

Reply:
Hmm I wonder. Can you do a little test to check whether that is the case? If it turns out to be helpful it'd be nice to have it in a separate commit with an explanation for what is going on.

Reply:
Analysis in https://github.com/cilium/ebpf/pull/649